### DISCUSSION TITLE
Strategy when a request fails or loads slowly

### DISCUSSION THREAD

**Discussion by writingdeveloper** [sindresorhus/got#1870] [Q&A]
I request data from the API server. However, this API server sometimes throws a connection wait exceeded error.

![Cap 2021-09-13 20-21-11-718](https://user-images.githubusercontent.com/6832586/133188226-7ae45db9-f7e7-4900-8ea6-6c887d7a9876.png)

I tested API server for 24hours and results are on below image. It shows It can access 67%~78% success range. 

It means when I request API 10 times, it only returns data 6~7 times.

I cannot control API server so I have to make strategy that when server cannot access to API server, is there any strategies to handle this problem?

I found [Retry API](https://github.com/sindresorhus/got/blob/main/documentation/7-retry.md#retry-api), and it seems that this API are suitable in my case.

I want to make it like this.

1. Request to API server
2. If it's not loaded 3 seconds after the request, how to request it additionally?

**Comment by szmarczak** [comment]
> If it's not loaded 3 seconds after the request, how to request it additionally?

You can set a `timeout` option. If it fails to load in X seconds, it will simply make a new retry unless the retry limit has been reached or the request is unsafe to retry.

**Reply by zeing** [reply]
how to retry when set timeout x seconds?
